### PR TITLE
chore(ci): set up tagged release against `rna/release` branch

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

Sets up tagged release against `rna/release` branch.

1. Developer opens a PR from `rna/main` -> `rna/release`
2. Developer merges the PR once e2e tests have passed on the PR
3. Merging the PR will publish `@aws-amplify/ui-react-native` to `@rna`.

#### Description of how you validated changes

Copy-pasted `next-release` logic.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/publish-rna.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+on:
+  push:
+    branches: [rna/release]
+
+jobs:
+  publish:
+    uses: ./.github/workflows/reusable-tagged-publish.yml
+    with:
+      dist-tag: rn-authenticator
```
Comment (wlee221) [inline .github/workflows/publish-rna.yml:15]: Should we just name this `rna`?

Comment (wlee221) [reply .github/workflows/publish-rna.yml:15]: Thought abbreviaiton `rna` was somewhat confusing for audiences without context, but open to thoughts.

Comment (calebpollman) [reply .github/workflows/publish-rna.yml:15]: Since this for internal testing, think `rna` is totally fine

Comment (wlee221) [reply .github/workflows/publish-rna.yml:15]:
```suggestion
      dist-tag: rna
```

**Review by calebpollman — APPROVED** [review]
LGTM 🚢

**Review by joebuono — APPROVED** [review]

**Pull request merged**